src/GitHub.VisualStudio/Services/UsageTracker.cs

```diff
@@ -194,21 +204,23 @@ void TimerTick(object sender, EventArgs e)
 
         async Task TimerTick()
         {
-            Debug.Assert(client != null, "TimerTick should not be triggered when there is no IMetricsService");
+            Initialize();
 
-            // Subsequent timer ticks should occur every 8 hours.
-            timer.Interval = TimeSpan.FromHours(8);
+            if (firstRun)
+            {
+                IncrementLaunchCount();
+                timer.Interval = TimeSpan.FromHours(8);
+                firstRun = false;
+            }
 
             if (userSettings == null)
```

Comment:
Well `TimerTick` will only ever get called on the main thread as it's triggered from a `DispatcherTimer`, so it was unnecessary before. But you're right that it's necessary now as `Initialize` may be called on a b/g thread from one of the `IncrementX` methods.